### DISCUSSION TITLE
Start date and End date facility for a schedule

### DISCUSSION THREAD

**Discussion by yogeshrao** [kagkarlsson/db-scheduler#475] [Ideas]
Checking if i can create a schedule which will start from X date and end at Y date ?

Requirement i am looking at is I would like to schedule a task which would recur every 4 days at 12 o clock from 1st of April.

Wondering if this is possible with db-scheduler ?

**Comment by kagkarlsson** [comment]
You can implement your own `Schedule`

**Reply by kagkarlsson** [reply]
End-dates is a bit more tricky. If you want to remove the task-instance on end-date, you have to create some sort of `CustomTask`. A "hack" would be to set next execution-time to far-future.

**Reply by barbosini** [reply]
If I write my own Schedule where the method getNextExecutionTime returns null if the end date is reached, will  the recurring task not execute anymore?

**Reply by barbosini** [reply]
Never mind my question. I've checked the code and the answer is no
Would it be possible to add support for end dates?
